src/main/java/ai/elimu/tasks/AllophoneUsageCountScheduler.java

```diff
@@ -41,15 +42,18 @@ public synchronized void execute() {
         // Integer = Usage count
         Map<Long, Integer> allophoneFrequencyMap = new HashMap<>();
 
-        // Summarize the usage count of each Word's Allophone based on the Word's usage count
+        // Summarize the usage count of each Word's Allophone based on the LetterToAllophoneMapping's 
+        // usage count (see LetterToAllophoneMappingUsageCountScheduler).
         List<Word> words = wordDao.readAllOrdered();
         logger.info("words.size(): " + words.size());
         for (Word word : words) {
-            for (Allophone allophone : word.getAllophones()) {
-                if (!allophoneFrequencyMap.containsKey(allophone.getId())) {
-                    allophoneFrequencyMap.put(allophone.getId(), word.getUsageCount());
-                } else {
-                    allophoneFrequencyMap.put(allophone.getId(), allophoneFrequencyMap.get(allophone.getId()) + word.getUsageCount());
+            for (LetterToAllophoneMapping letterToAllophoneMapping : word.getLetterToAllophoneMappings()) {
+                for (Allophone allophone : letterToAllophoneMapping.getAllophones()) {
+                    if (!allophoneFrequencyMap.containsKey(allophone.getId())) {
+                        allophoneFrequencyMap.put(allophone.getId(), letterToAllophoneMapping.getUsageCount());
+                    } else {
+                        allophoneFrequencyMap.put(allophone.getId(), allophoneFrequencyMap.get(allophone.getId()) + letterToAllophoneMapping.getUsageCount());
+                    }
```

Comment:
Nitpick: You can replace if..else with
`allophoneFrequencyMap.put(allophone.getId(), allophoneFrequencyMap.getOrDefault(allophone.getId(), 0) + letterToAllophoneMapping.getUsageCount());
`
https://www.geeksforgeeks.org/hashmap-getordefaultkey-defaultvalue-method-in-java-with-examples/

Reply:
@Keerthi4308 Thanks for the suggestion 🙂

The less code, the better. So this would be a good improvement!

To avoid multiple types of changes in one pull request, I've created a separate issue for implementing your suggestion: https://github.com/elimu-ai/webapp/projects/7#card-61392765